src/services/anchor-service.ts

```diff
@@ -139,19 +160,33 @@ export class AnchorService {
 
     if (candidates.length === 0) {
       logger.imp('No candidates found. Skipping anchor.')
-      return {
-        acceptedRequestsCount: groupedRequests.acceptedRequests.length,
-        alreadyAnchoredRequestsCount: groupedRequests.alreadyAnchoredRequests.length,
-        anchoredRequestsCount: 0,
-        conflictingRequestCount: groupedRequests.conflictingRequests.length,
-        failedRequestsCount: groupedRequests.failedRequests.length,
-        failedToPublishAnchorCommitCount: 0,
-        unprocessedRequestCount: groupedRequests.unprocessedRequests.length,
-        candidateCount: candidates.length,
-        anchorCount: 0,
-      }
+      logAnchorSummary(groupedRequests, candidates)
+      return
     }
 
+    try {
+      const results = await this._anchorCandidates(candidates)
+      logAnchorSummary(groupedRequests, candidates, results)
+      return
+    } catch (err) {
+      // TODO(NET-1623): Add alert that something went wrong and we are retrying
+      logger.warn(
+        `Updating PROCESSING requests to FAILED so they are retried in the next batch because an error occured while creating the anchors: ${err}`
```

Comment:
Why FAILED and not PENDING if the plan is to retry them?

Reply:
Well at this point PENDING + FAILED both get retried immediately and I do not have any strong preference for either. I can see how PENDING might make more sense when trying to understand everything so I'll change it to that.

Reply:
Yeah I thought the plan was to eventually stop retrying FAILED requests and only mark requests as FAILED when we know they have truly failed and won't recover on a retry.  Then for things we want to retry we move them back to PENDING.  Did you have a different end state in mind?

Reply:
Ah wait after revising it I see why I changed them as FAILED. Because I include them in the `failedRequestCount` in the logged `anchorSummary`. I'll still include them in the `failedRequestCount` even if they weren't marked FAILED cause technically they did fail.

Reply:
> Then for things we want to retry we move them back to PENDING. Did you have a different end state in mind?

We will probably introduce so more complicated retry logic later when things get simplified and separated. Probably a RETRY state or something like that. But thats for later.

Reply:
Should we split out `failedRequestCount` into two separate counters?  One for permanently failed requests and one for requests that failed but will be retried?  We can add or remove entries from the anchorSummary log message as we want.

Reply:
Current:
`failedRequestCount` = purposefully marked failed requests + all requests that are being reverted to PENDING

Your proposal: 
`failedRequestCount` = purposefully marked failed requests 
`permanentlyFailedRequest` = `conflictingRequestCount` (already tracked)
`failedButRetriableCount` = `failedRequestCount` - `conflictingRequestCount` + all requests that are being reverted to PENDING (note that retriable does not mean the request will be retried. If the request expired that it will not)

Based on what is in the anchor summary in the current situation we can infer the amount of `failedButRetriableCount` through `failedRequestCount - conflictingRequestCount`. 

Questions:
- The additional information your proposal gives is the amount of requests that we have purposefully marked as failed during our processes. Is this information what we want? 
- Do we also want the number of requests that are being reverted to PENDING? 
- Are we interested in have a specific label for `retriableRequestsCount` even if it can be inferred for clarity? And if so is it implicit knowledge that some of these requests won't be retried because they have expired in the next batch

Feel free to ping me or hop in a chat if this is confusing.

Reply:
I didn't realize we could infer the difference from what we already have.  I don't feel strongly about what we do here - I think we should do whatever is simplest, and we shouldn't be afraid to redefine some of these fields in the log summary if it helps with long-term clarity and simplicity.

Reply:
Based on this exchange i think i'll add a label for `retriableRequestsCount`. I can add a comment explaining this there as well for clarity so we don't forget